[TF] Further cleanup of CompilerRuntime and friends

burmako opened the pull request:
I was getting started with TPU support, and this functionality wasn't doing what it was advertising. After a chat with the team, I learned that it was designed to work with GPE, and with GPE gone it doesn't make much sense anymore. Therefore, I decided to go ahead and remove it.

burmako committed:
Remove _ExecutionMode (TF-517)

  * TF_CreateConfig.enable_xla_compilation is now hardcoded to 0.
  * TPU tests now unconditionally fail instead of pretending to work.

burmako committed:
Remove enableTPU, enableGPU and enableCPU

These functions have empty bodies, but they have pretty high-profile
names which gives an impression that they do something important.

With that in mind, I think it would be best to remove these functions
without any deprecation notice because there aren't any comparable
replacements at the moment.

burmako commented:
@swift-ci Please test tensorflow

burmako commented:
@swift-ci Please test tensorflow

rxwei commented on stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift:
```diff
@@ -91,9 +90,7 @@ extension TestSuite {
   public func testTPU(_ name: String, _ body: @escaping () -> Void) {
 #if TPU
     test(name + "_TPU") {
```
I think all these test functions are also obsolete actually.

burmako replied:
Further cleanup, here I come!

burmako replied:
@rxwei I've done some further cleanup. PTAL.

burmako committed:
Remove obsolete remnants of the TPU testing infrastructure (TF-520)

This commit removes the remnants of the obsolete TPU testing
infrastructure. In the nearest future, we are planning to restore it
within the new tracing runtime (https://bugs.swift.org/browse/TF-520).

burmako committed:
Remove obsolete remnants of the TPU runtime infrastructure (TF-519)

This commit removes the remnants of the obsolete TPU runtime
infrastructure. In the nearest future, we are planning to restore it
within the new tracing runtime (https://bugs.swift.org/browse/TF-519).

burmako commented:
@swift-ci Please test tensorflow

burmako commented:
@swift-ci Please test tensorflow

burmako commented:
@swift-ci I'm begging you please test tensorflow

bgogul reviewed (approved):
Thanks, Eugene!

dan-zheng commented:
@swift-ci Please test tensorflow

dan-zheng commented:
@swift-ci Please test tensorflow

burmako commented:
Merging while the macOS job is running, since it is expected that this job will fail anyway.
